src/NoteWorkspace.ts

```diff
@@ -559,7 +560,12 @@ export class NoteWorkspace {
     } else {
       // create the file if it does not exist
       const contents = NoteWorkspace.newNoteContent(noteTitle);
-      writeFileSync(filepath, contents);
+      const edit = new vscode.WorkspaceEdit();
+      const fileUri = vscode.Uri.file(filepath);
+      edit.createFile(fileUri);
+      vscode.workspace.applyEdit(edit).then(() => 
```

Comment:
I see here there is opportunity for a callback. I was thinking I wonder if there's a way to somehow leverage this when using goto definition for a non-existent note, ie here:

https://github.com/kortina/vscode-markdown-notes/blob/master/src/MarkdownDefinitionProvider.ts#L39

In the case where we must create the note, can we await the async vscode create file operation before we return the vscode.Location array in this function?

What do you think?